Mathlib/Algebra/Group/Defs.lean

```diff
@@ -512,15 +610,28 @@ instance (priority := 100) CancelCommMonoid.toCancelMonoid (M : Type u) [CancelC
     CancelMonoid M :=
   { mul_right_cancel := fun a b c h => mul_left_cancel <| by rw [mul_comm, h, mul_comm] }
 
+/-- Any `CancelMonoid G` satisfies `IsCancelMul G`. -/
+instance (priority := 100) CancelMonoid.to_IsCancelMul (M : Type u) [CancelMonoid M] :
+    IsCancelMul M :=
+  { mul_left_cancel := LeftCancelSemigroup.mul_left_cancel
+    mul_right_cancel := RightCancelSemigroup.mul_right_cancel }
+
+--Porting note: the `defLemma` linter complains about this one if it is generated by `to_additive`.
+/-- Any `AddCancelMonoid G` satisfies `isCancelAdd G`. -/
+instance (priority := 100) AddCancelMonoid.to_isCancelAdd (M : Type u) [AddCancelMonoid M] :
+    isCancelAdd M :=
+  { add_left_cancel := AddLeftCancelSemigroup.add_left_cancel
+    add_right_cancel := AddRightCancelSemigroup.add_right_cancel }
+
```

Comment:
even is `[to_additive]` won't generate it, I think you still need to
```
attribute [to_additive AddCancelMonoid.to_isCancelAdd] CancelMonoid.to_IsCancelMul
```

Writing that out also identifies that one is PascalCase and the other is camelCase to the right of `to_`.